DiagonalMatrix/DiagonalMatrix.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace DiagonalMatrixNamespace
+{
+    public class DiagonalMatrix
+    {
+        public int[] DiagonalElements { get; set; }
+        public int Size { get; }
+
+        public DiagonalMatrix(params int[] matrixElements)
+        {
+            if (matrixElements is null)
+            {
+                DiagonalElements = [];
+                Size = 0;
+            }
+            else
+            {
+                Size = matrixElements.Length;
+                DiagonalElements = new int[Size];
+                Array.Copy(matrixElements, DiagonalElements, Size);
+            }
+        }
+
+        public int this[int i, int j]
+        {
+            get => IsElementOnMainDiagonal(i, j) ? DiagonalElements[i] : 0;
+            set
+            {
+                if (IsElementOnMainDiagonal(i, j))
+                {
+                    DiagonalElements[i] = value;
+                }
+            }
+        }
+
+        public int Track()
+        {
+            var diagonalMatrixElementsSum = 0;
+
+            foreach (var element in DiagonalElements)
+            {
+                diagonalMatrixElementsSum += element;
+            }
+
+            return diagonalMatrixElementsSum;
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is DiagonalMatrix matrixToCompare && matrixToCompare.Size == Size)
+            {
+                for (int i = 0; i < Size; i++)
+                {
+                    if (DiagonalElements[i] != matrixToCompare.DiagonalElements[i])
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            int hash = 17;
+
```

Comment:
too time consuming implementation of hash, it has to work faster then Equals, but I think that in this case it is not true)))

Reply:
Fixed. Made the numbers lower and removed the GetHashCode() method calling for every element of the array. Hope that will optimize the performance enough. In case that is not enough, I would ask for a hint, if possible